src/System.Runtime.Analyzers/UnitTests/AvoidZeroLengthArrayAllocationsTests.cs

```diff
@@ -34,7 +34,8 @@ public void EmptyArrayCSharp()
                 @"namespace System { public class Array { public static T[] Empty<T>() { return null; } } }";
 
             const string badSource = @"
-[System.Runtime.CompilerServices.Dynamic(new bool[0])] // no
```

Comment:
Ok, restored that with a comment that it's intentional.

I'm thinking about a new PR, which adds a boolean parameter `allowCompileErrors` to `VerifyXXX` methods, that defaults to `false`. The dozens of tests that currently contain non-compiling code I will update to explicitly pass `true`. This would prevent new tests with accidental non-compiling code.

Any thoughts/objections on that?